### PR TITLE
enh: mesh smoothing ability

### PR DESCRIPTION
**Pull request by StolkArjen**
As 'discussed' in bug 3266. :)
http://bugzilla.fieldtriptoolbox.org/show_bug.cgi?id=3266

The functionality was there already, just not accessible top-down. Esthetically, the difference may be significant, see screenshot in corresponding bug report.

### CODE REVIEW

File: private/prepare_mesh_headshape.m [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
 % get the specific options
 cfg.headshape    = ft_getopt(cfg, 'headshape');
+cfg.smooth       = ft_getopt(cfg, 'smooth');   % no default
```
Comment (robertoostenveld) [inline private/prepare_mesh_headshape.m]: ok, ok. But please do add it to the help as well.

File: private/prepare_mesh_headshape.m [hunk 2]
```diff
@@ -95,6 +96,13 @@
...
+% smooth the mesh
+if ~isempty(cfg.smooth)
+  for i=1:nmesh
+    [headshape(i).pos,headshape(i).tri] = fairsurface(headshape(i).pos, headshape(i).tri, cfg.smooth);
+  end
+end
+
 % the output should only describe one or multiple boundaries and should not
 % include any other fields
 mesh = rmfield(headshape, setdiff(fieldnames(headshape), {'pos', 'tri'}));
```
Comment (robertoostenveld) [inline private/prepare_mesh_headshape.m]: can you please change this into 
```
mesh = keepfield(headshape, {'pos', 'tri'});
```

Comment (StolkArjen) [inline private/prepare_mesh_headshape.m]: Thanks. It's done. Btw, it's keepfieldS.

**Comment by StolkArjen** [comment]
Ready to pull

**Pull request merged**